src/condor_tools/htcondor_cli/job.py

```diff
@@ -76,6 +81,41 @@ def __init__(self, logger, submit_file, **options):
                 submit_data = f.read()
             submit_description = htcondor.Submit(submit_data)
 
+            annex_name = options["annex_name"]
+            if annex_name is not None:
+                # Tag the job for use by `htcondor annex create` later.
+                submit_description["MY.TargetAnnexName"] = f'"{annex_name}"'
+
+                # Require that this job run only on the specified annex,
+                # whose name we'll verify by requiring that "the same"
+                # user requested the annex in question.
+                #
+                # We don't need to know what htcondor.Submit.queue() thinks
+                # what the Owner or Submitter of the job will be, because the
+                # AuthenticatedIdentity is determined by the tokens that we
+                # hand out.  We will hand them out in such a way that this
+                # code always gets the right answer.
+                username = getpass.getuser()
+                my_identity = f'{username}@annex.osgdev.chtc.io'
+                # This looks awful, but AuthenticatedIdentity isn't set in
+                # the startd's copy of the ad.
+                annex_requirements = f'((ifthenelse(TARGET.AuthenticatedIdentity is undefined, true, "{my_identity}" == TARGET.AuthenticatedIdentity)) && (MY.TargetAnnexName == TARGET.AnnexName))'
+
+                # FIXME: htcondor.Submit.get() should allow its second
+                # argument to be None.
+                # requirements = submit_description.get("requirements", "")
+                # FIXME: htcondor.Submit object should be iterable so that
+                # `if key in s` works.
+                if "requirements" in submit_description.keys():
+                    submit_description["requirements"] = f'({submit_description["requirements"]}) && ({annex_requirements})'
+                else:
+                    submit_description["requirements"] = annex_requirements
+
+                # Flock to the annex CM.
+                # FIXME: This should probably come from configuration.
+                collector = "htcondor-cm-hpcannex.osgdev.chtc.io"
```

Comment:
So the `htcondor.param` object does support the `get` method so we could at least do something like:
```suggestion
                collector = htcondor.param.get("ANNEX_COLLECTOR", "htcondor-cm-hpcannex.osgdev.chtc.io")
```
Assuming that `ANNEX_COLLECTOR` is a name we'd consider asking people to put in their configs, and assuming we're ok defaulting to a hardcoded value until people actually do edit their configs. (Don't actually commit this suggestion.)

Reply:
This is almost exactly what I had in mind.  I'd have to think about making it `HPC_ANNEX_COLLECTOR`, but I'm also tempted by the idea that leaving it `ANNEX_COLLECTOR` will make it more likely that I update the cloud annex to make use of the newer technologies here.

I like the idea of using an in-code default here rather than adding `ANNEX_COLLECTOR` to the param table.